File: 1673.c
#include <stdio.h>

int main(void)
{
	int n = 0, k = 0;

	freopen("input.txt", "r", stdin);

	while(scanf("%d%d", &n, &k) == 2) {
		int stamp = 0;
		int res = 0;
		while(n) {
			res = res + n;
			stamp += n;
			n = stamp / k;
			stamp -= n*k;
		}
		printf("%d\n", res);
	}

	return 0;
}

File: _1000/1673.py
import sys
sys.stdin = open('input.txt')
input = sys.stdin.readline

for line in sys.stdin:
    n, k = map(int, line.split())
    res = 0
    stamp = 0
    while n != 0:
        res += n
        stamp += n
        n = stamp//k
        stamp %= k
    print(res)
